controllers/activemqartemis_controller_test.go

```diff
@@ -91,6 +91,401 @@ var _ = Describe("artemis controller", func() {
 		}
 	})
 
+	Context("Affinity Test", func() {
+		It("setting Pod Affinity", func() {
+			By("Creating a crd with pod affinity")
+			ctx := context.Background()
+			crd := generateArtemisSpec(namespace)
+			labelSelector := metav1.LabelSelector{}
+			labelSelector.MatchLabels = make(map[string]string)
+			labelSelector.MatchLabels["key"] = "value"
+
+			podAffinityTerm := corev1.PodAffinityTerm{}
+			podAffinityTerm.LabelSelector = &labelSelector
+			podAffinity := corev1.PodAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+					podAffinityTerm,
+				},
+			}
+			crd.Spec.DeploymentPlan.Affinity.PodAffinity = &podAffinity
+
+			By("Deploying the CRD " + crd.ObjectMeta.Name)
+			Expect(k8sClient.Create(ctx, &crd)).Should(Succeed())
+
+			createdCrd := &brokerv1beta1.ActiveMQArtemis{}
+			createdSs := &appsv1.StatefulSet{}
+
+			By("Making sure that the CRD gets deployed " + crd.ObjectMeta.Name)
+			Eventually(func() bool {
+				return getPersistedVersionedCrd(crd.ObjectMeta.Name, namespace, createdCrd)
+			}, timeout, interval).Should(BeTrue())
+			Expect(createdCrd.Name).Should(Equal(crd.ObjectMeta.Name))
+			By("Checking that Stateful Set is Created with the node selectors " + namer.CrToSS(createdCrd.Name))
+			Eventually(func() bool {
+				key := types.NamespacedName{Name: namer.CrToSS(createdCrd.Name), Namespace: namespace}
+
+				err := k8sClient.Get(ctx, key, createdSs)
+
+				if err != nil {
+					return false
+				}
+				return createdSs.Spec.Template.Spec.Affinity.PodAffinity != nil
+			}, timeout, interval).Should(Equal(true))
+
+			By("Making sure the pd affinity are correct")
+			Expect(createdSs.Spec.Template.Spec.Affinity.PodAffinity.RequiredDuringSchedulingIgnoredDuringExecution[0].LabelSelector.MatchLabels["key"] == "value").Should(BeTrue())
+
+			By("Updating the CR")
+			Eventually(func() bool { return getPersistedVersionedCrd(crd.ObjectMeta.Name, namespace, createdCrd) }, timeout, interval).Should(BeTrue())
```

Comment:
for update, you need to get/apply/update to be in the Eventually, such that you are applyiing your changes to the current version. You are competing here with the operator status update that will modify the ResourceVersion.

Here you are getting a version of the crd, but it may not accept your update and may fail, or get overwritten by the operator version.